src/v1/device_manager_client.ts

```diff
@@ -1788,6 +1810,18 @@ export class DeviceManagerClient {
               res.on('end', () => {
                 const deviceConfig: protos.google.cloud.iot.v1.IDeviceConfig =
                   JSON.parse(data);
+                if (isBinaryDataFormat()) {
+                  deviceConfig.cloudUpdateTime = timeSecondsNanos(
+                    JSON.parse(data).cloudUpdateTime
```

Comment:
instead of parsing again, just reuse `deviceConfig` like so:
```
                  deviceConfig.cloudUpdateTime = timeSecondsNanos(
                    deviceConfig.cloudUpdateTime)
```

Reply:
I would just use a type assertion for now - 
```
deviceConfig.cloudUpdateTime = timeSecondsNanos(
                    deviceConfig.cloudUpdateTime as string
                  );
```